tensorflow_datasets/scripts/random_image_generator.py

```diff
@@ -0,0 +1,107 @@
+
+""" Generate random images that compress better.
+By replacing the larger images with more compressable equivalents."""
+
+import os
+from zipfile import ZipFile
+from PIL import Image
+import numpy as np
+import tarfile
+import zipfile
+import shutil
+import gzip
+
+
+# images_path variable to store the path of images for later compression
+images_path = []
+
+
+def image_process(root):
+    """ Generate random images and remove noise of the
+        images to compress better.
+
+          Args:
+           root: path of the images to get processed
+
+    """
+    images_path.append(root)
+    image = np.array(Image.open(root))
+    image = np.ones_like(image%255) * np.random.randint(np.amax(image%255))
+    image = Image.fromarray(image)
+    image = image.convert('RGB')
+    image.save(root)
+
+
+def rewrite_zip(filepath, filename):
+    """ Rewrite the older .zip files into new better compressed one
+
+        Args:
+            filepath: path of the older zip file that need to be compressed
+            filename: name of the older zip file that need to be compressed
+
+    """
+     # creating a temporary file to store images 
+    with tempfile.TemporaryDirectory(dir=filepath) as temp:
+
+      # Extraction of compressed .zip file
+      with ZipFile(filepath+filename, 'r') as zip:
+          zip.extractall(path=temp.name)
+          print(temp.name)
+
+          process_dir(temp.name+"/")  # Image Processing
+
+      # Compressed the .zip file again
+      with ZipFile(filepath+filename, 'w') as zip:
+          for file in images_path:
```

Comment:
* If there is multiple zip files, the second zip files will compress all recorded images, including the ones which do not belong to the zip file.
* Also if the zip file was containing other files, they are lost when compressing the zip files.